src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -737,20 +736,24 @@
 
 (schema/defn ^:always-validate
   v4-catalog-fn :- IFn
-  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)]
+  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
+   current-code-id-fn :- IFn]
   (fn [request]
     (let [request-options (decode-catalog-post-body (slurp (:body request)))]
       {:status 200
        :headers {"Content-Type" "application/json"}
        :body (json/encode
               (jruby-protocol/compile-catalog jruby-service
                                               (:jruby-instance request)
-                                              request-options))})))
+                                              (assoc request-options
+                                                     "code_id"
+                                                     (current-code-id (jruby-request/get-environment-from-request request)))))})))
```

Comment:
I think you have one too many parentheses on the end of this line.

Reply:
oh, it's because it is `current-code-id-fn` instead of `current-code-id`.